src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptCompletionProvider.cs

```diff
@@ -15,12 +15,13 @@ internal abstract class VSTypeScriptCompletionProvider : CompletionProvider
         public sealed override bool ShouldTriggerCompletion(SourceText text, int caretPosition, CompletionTrigger trigger, OptionSet options)
         {
             Debug.Fail("For backwards API compat only, should not be called");
-            return ShouldTriggerCompletionImpl(text, caretPosition, trigger, CompletionOptions.Default.TriggerOnTypingLetters);
+            var defaultOptions = CompletionOptions.Default;
+            return ShouldTriggerCompletionImpl(text, caretPosition, trigger, defaultOptions.TriggerOnTypingLetters, defaultOptions.ExpandedCompletionBehavior);
         }
 
         internal sealed override bool ShouldTriggerCompletion(HostLanguageServices languageServices, SourceText text, int caretPosition, CompletionTrigger trigger, CompletionOptions options, OptionSet passThroughOptions)
-            => ShouldTriggerCompletionImpl(text, caretPosition, trigger, options.TriggerOnTypingLetters);
+            => ShouldTriggerCompletionImpl(text, caretPosition, trigger, options.TriggerOnTypingLetters, options.ExpandedCompletionBehavior);
 
-        protected abstract bool ShouldTriggerCompletionImpl(SourceText text, int caretPosition, CompletionTrigger trigger, bool triggerOnTypingLetters);
+        protected abstract bool ShouldTriggerCompletionImpl(SourceText text, int caretPosition, CompletionTrigger trigger, bool triggerOnTypingLetters, ExpandedCompletionMode completionMode);
```

Comment:
I think this would require a dual insertion if you add a parameter here.  I can't see a super easy way to get around it though.

Reply:
Correct. I already know which would be the changes to be performed from the TypeScript side.

Reply:
An alternative would be to modify the context that's passed in [this method](https://github.com/dotnet/roslyn/blob/d8862d664fdebaf5aefb3ecc0284712f79bc0338/src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs#L90-L103) to have an `ExpandedCompletionBehavior` of `ExpandedCompletionMode.ExpandedItemsOnly`. I'm not sure if these will be carried all the way down to the `CompletionOptions` of the context that's passed in [here](https://github.com/dotnet/roslyn/blob/d8862d664fdebaf5aefb3ecc0284712f79bc0338/src/Features/Core/Portable/Completion/CompletionProvider.cs#L24-L27).

Reply:
We can avoid dual insertion: https://github.com/dotnet/roslyn/pull/60635#discussion_r845601632